### PR TITLE
Setup action for CS Fixer

### PR DESCRIPTION
**Pull request by jdarwood007**
This will now run PHP-CS-Fixer as a dedicated GitHub Action.  When it finds errors with the changed files, it will fail the check.

This is almost a copy and paste from: https://github.com/OskarStark/php-cs-fixer-ga

### CODE REVIEW

**Comment by jdarwood007** [comment]
FYI, it ran on this PR as well if you want to see the check details.

**Comment by jdarwood007** [comment]
Additionally,  Just added a custom cs fixer that does the fixing of the line endings, replacing the need for the EOL script.  As a bonus, it means it can also run locally easily for devs.

File: .github/phpcs/SMFClosingTag.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		// We have closing bracket then closing barcket, single white space and then closing tag. Add one more return.
+		if ($count > 0 && $tokens[$count - 1]->isGivenKind(T_CLOSE_TAG) && $tokens[$count - 2]->isGivenKind(T_WHITESPACE) && $tokens[$count - 3]->getContent() !== ';') {
+			$tokens[$count - 2] = new Token([
+				T_WHITESPACE,
+				$this->whitespacesConfig->getLineEnding() . $this->whitespacesConfig->getLineEnding()
+			]);
+		}
+
+		// We have a closing bracket, and then closing tag, no white space, add returns.
+		// There is no ID/Name for closing curely bracket or semi-colon.
```
Comment (tyrsson) [inline .github/phpcs/SMFClosingTag.php:105]: Not sure if this is what you're looking for but check out T_CLOSE_CURLY_BRACKET

Comment (jdarwood007) [reply .github/phpcs/SMFClosingTag.php:105]: I tried, it doesn't exist.  I checked getId and getName for the char and both came back with nothing.

Comment (tyrsson) [reply .github/phpcs/SMFClosingTag.php:105]: Yea, appears to be some custom defined const. I just remember seeing it, I think maybe php codesniffer. I guess you could always define it if need be.

**Review by Sesquipedalian — APPROVED** [review]

**Comment by Sesquipedalian** [comment]
Thanks for doing this, @jdarwood007

**Pull request merged**